lib/action_subscriber/middleware/error_handler.rb

```diff
@@ -8,25 +8,18 @@ def initialize(app)
       end
 
       def call(env)
-        job_mutex = ::Mutex.new
-        job_complete = ::ConditionVariable.new
+        @app.call(env)
+      rescue => error
+        logger.error "FAILED #{env.message_id}"
 
-        job_mutex.synchronize do
-          ::Thread.new do
-            job_mutex.synchronize do
-              begin
-                @app.call(env)
-              rescue => error
-                logger.error "FAILED #{env.message_id}"
-                ::ActionSubscriber.configuration.error_handler.call(error, env.to_h)
-              ensure
-                job_complete.signal
-              end
-            end
-          end
-
-          # TODO we might want to pass a timeout to this wait so we can handle jobs that get frozen
-          job_complete.wait(job_mutex)
+        # There is more to this rescue than meets the eye. MarchHare's java library will rescue errors
```

Comment:
10/10 will buy you at least 2 🌯 s the next time you are in town for leaving this comment